python/cudf/cudf/core/dataframe.py

```diff
@@ -2392,9 +2392,6 @@ def join(
             )
             method = type
 
-        if how not in ["left", "right", "inner", "outer"]:
-            raise NotImplementedError("unsupported {!r} join".format(how))
-
         if how == "right":
```

Comment:
Should we do similar to `rightsemi` and `rightanti`?

Reply:
My plan for this is to alias these in a join cleanup PR so that they just swap the left and right hand sides of the join and then use the same codepath as `leftanti` and `leftsemi`. We might need to special case it so that it returns the columns in right order at the end but it shouldn't be too complicated.